frontend/src/components/CodeIDE/CodeIDEButtons.tsx

```diff
@@ -190,13 +205,36 @@ export const CodeIDEButtons = (props: CodeIDEButtonProps) => {
               <MenuList>
                 {curIdx !== 0 && (
                   <>
-                    <MenuItem onClick={renameFunc}>Rename</MenuItem>
-                    <MenuItem onClick={reload}>Reload</MenuItem>
-                    <MenuItem onClick={deleteFunc}>Delete</MenuItem>
+                    <MenuItem icon={<EditIcon />} onClick={renameFunc}>
+                      Rename
+                    </MenuItem>
+                    <MenuItem icon={<RepeatIcon />} onClick={reload}>
+                      Reload
+                    </MenuItem>
+                    <MenuItem icon={<DeleteIcon />} onClick={deleteFunc}>
+                      Delete
+                    </MenuItem>
                   </>
                 )}
-                <MenuItem>Share</MenuItem>
-                <MenuItem>Download</MenuItem>
+                <Tooltip
+                  placement="top"
+                  label={
+                    loggedIn
+                      ? curIdx === 0
+                        ? 'Save your code!'
+                        : ''
+                      : 'Login and save your code!'
```

Comment:
Should add something to indicate that those are prerequisites to sharing code (e.g. "Save your code first!" or "Save your code to enable sharing!"). Otherwise it reads a bit like pressing that button will save your code.

Reply:
Updated